### DISCUSSION TITLE
How to detect transparency on an image asset?

### DISCUSSION THREAD

**Discussion by tomkiss** [craftcms/cms#12520] [Q&A] [answered]
I'm trying to detect whether an image asset has transparency, in a twig template.

Looking at [craft\base\Image][1] I thought that [`isTransparent`][2] may be able to help with this.

So to test it, I created a small Module with a PHP function, available as a Twig filter, to load the asset and return the value from `isTransparent`, something like this:

```php    
    public function getIsTransparent(Asset $asset) 
    {
        if (!isset($asset)) {
            return null;
        } else {
            $images = Craft::$app->getImages();
            $localCopy = TransformHelper::getLocalImageSource($asset);
            $image = $images->loadImage($localCopy, true);
            return $image->getIsTransparent();
        }
    }
```

...but no matter what image is loaded, it *always* returns 1.

However, if I create a new ImageMagick instance and test the image like so:

```php    
    public function getIsTransparent(Asset $asset) 
    {
        if (!isset($asset)) {
            return null;
        } else {
            $localCopy = TransformHelper::getLocalImageSource($asset);
            $imgk = new Imagick();
            $imgk->readImage($localCopy);
            return $imgk->getImageAlphaChannel() ? 1 : 0;
        }
    }
```

This second solution works as expected.

I've solved the issue I had, but I can't help thinking I'm doing something wrong here with the [`isTransparent`][2] property. Any pointers much appreciated!

  [1]: https://docs.craftcms.com/api/v4/craft-base-image.html
  [2]: https://docs.craftcms.com/api/v4/craft-base-image.html#istransparent

**Comment by tomkiss** [comment, marked as answer]
Note that I've since discovered that isTransparent doesnt appear to work in craft 4.3.6, but does in 4.3.5. As such, I've filed a [bug report](https://github.com/craftcms/cms/issues/12565).

**Reply by brandonkelly** [reply]
And now the bug is fixed!
